src/CoreWCF.Primitives/src/CoreWCF/Description/TypeLoader.cs

```diff
@@ -1117,9 +1119,73 @@ private void VerifyConsistency(OperationConsistencyVerifier verifier)
                     requestDescription.Body.WrapperName = requestDescription.Body.WrapperNamespace = null;
                 }
             }
+
+            var serviceImplementationMethodInfo = FindServiceImplementationMethodInfo(operationDescription, contractDescription);
+            if (serviceImplementationMethodInfo == null)
+            {
+                // we do care only about Authorize attributes that decorate non callback contract
+                Fx.AssertAndThrow(direction == MessageDirection.Output,
+                    $"Unable to find service implementation in contract {contractDescription.Name}");
+            }
+            else
+            {
+                methodAttributes = serviceImplementationMethodInfo.GetCustomAttributes(false);
+                foreach (var methodAttribute in methodAttributes.OfType<IAuthorizeData>())
+                {
+                    operationDescription.AuthorizeData.Add(methodAttribute);
+                }
+            }
+
             return operationDescription;
         }
 
+        private static MethodInfo FindServiceImplementationMethodInfo(OperationDescription operationDescription, ContractDescription contractDescription)
+        {
+            Type serviceType = typeof(TService);
+            MethodInfo[] methodInfos = serviceType.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+            // The source generator will produce an method with identical name but with attrributes marked with either CoreWCF.Injected or FromServices attributes.
+            // We need to to fetch AuthorizeAttribute from user provided implementation.
```

Comment:
I think we should have the source generator copy all  attributes across to the generated method. We have the other existing attribute for AuthorizeRoleAttribute to provide equivalent behavior to PrincipalPermssionAttribute in NetFx. Doing this in the generator feels like the right way to do this too.

Reply:
done in https://github.com/CoreWCF/CoreWCF/pull/850/commits/a2c8dbc237534bd8922cfcb7f2b2759694841cf7